### ISSUE TITLE
Installation Manager

### ISSUE THREAD

**Issue by An-Rcc** [platformio/platformio-core-installer#1743]
%23 Description of problem
  Leave a comment...

  BEFORE SUBMITTING, PLEASE SEARCH FOR DUPLICATES IN
  - https://github.com/platformio/platformio-vscode-ide/issues%3Fq=is%3Aissue 

  %23 Configuration

  VSCode: 1.61.0
  PIO IDE: v2.3.3
  System: Windows_NT, 10.0.22471, x64

  %23 Exception
  ```
  Error: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received null
	at c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:771390
	at a (c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:771284)
	at w (c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:771035)
	at u (c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:770559)
	at c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:771343
	at new Promise (<anonymous>)
	at l (c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:771322)
	at e.callInstallerScript (c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:7215)
	at async H.value (c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\node_modules\platformio-node-helpers\dist\index.js:1:763673)
	at async t.default.install (c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\dist\extension.js:1:6773)
	at async c:\Users\An\.vscode\extensions\platformio.platformio-ide-2.3.3\dist\extension.js:1:12776
  ```

**Comment by svenvucic** [comment]
> Hi @An-Rcc ,
> 
> This is Ivan from the PlatformIO team. We have a lot of similar reports and can't reproduce them on our virtual machines. Could you help us to debug this issue? Please contact [me@ikravets.com](mailto:me@ikravets.com).

Hi @ivankravets , @An-Rcc,

I've had the same issue as the 90% of the people with the "Installation Manager" issue name.

First of all, info from mine VSCode:
Version: 1.61.1 (user setup)
Commit: c13f1abb110fc756f9b3a6f16670df9cd9d4cf63
Date: 2021-10-14T01:19:42.089Z
Electron: 13.5.1
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 6.1.7601 (Windows 7 Pro x64 user here.)

These are the steps I did to get PlatformIO working (it did work normally in VS Code before I updated something in the extension.

1. Installed the latest Python for Windows 7 x64 (this is version python-3.8.8-amd64.exe) - I also had installed Python2.7.18 (but latest PlatformIO does not work with Python 2 so I think this is not relevant for this issue
2. Put path to Python38 folder from point 1 into Windows USER environment variable PATH (installation didn't do it for some reason)
3. I already had installed PlatformIO 2.3.3 so I had the folder C:\Users\<user>\.platformio where I deleted everything in folder
C:\Users\<user>\.platformio\python3 and copied the folders and files from Python38 folder (point 1)
4. Removed the checkbox from PlatformIO Extension Settings for property "Platformio-ide: Use Builtin Python"

Restarted VS Code, waited for PlatformIO to finish its thing and from then on - everything worked as it should! My opinion is that point under number 3 is what really helped, but since I did all the points without opening VS Code in between, I can't be sure.

Hope it helps!

**Comment by ivankravets** [comment]
@svenvucic do you mean that our pre-built Python does not work on your machine? Could you re-test https://dl.registry.platformio.org/download/platformio/tool/python-portable/1.30902.0/python-portable-windows_amd64-1.30902.0.tar.gz ?

**Comment by svenvucic** [comment]
@ivankravets 
Ah, as it is always with software- now I can't reproduce the problem anymore, even from the clean installation of VS Code and platformIO. The only thing I didn't try is to remove python3.8.8 and python2.7. since I didn't have more time today.

When I replaced python with the one you provided - VS Code was still working fine (but it was working fine from yesterday evening when I managed to resolve the issue and write the message above). Will get back to you on Monday with more testing!

**Comment by ivankravets** [comment]
Fixed in [PlatformIO IDE for VSCode v2.3.4](https://marketplace.visualstudio.com/items/platformio.platformio-ide/changelog). Please update PlatformIO IDE extension to the 2.3.4 version and restart VSCode.

Does it work for you now?

**Comment by svenvucic** [comment]
> Fixed in [PlatformIO IDE for VSCode v2.3.4](https://marketplace.visualstudio.com/items/platformio.platformio-ide/changelog). Please update PlatformIO IDE extension to the 2.3.4 version and restart VSCode.
> 
> Does it work for you now?

I've uninstalled every python that I had on my PC, removed VS Code including (.vscode, .platformio and code folders from Users/ and Users/Roaming folders). So, let's say I've had a clean start.

Installed a fresh VS Code, fresh Platformio (2.3.4) + arduino and espressif frameworks. **Everything works**.
I think it is better to ask the rest of the guys that had the same issue, since I've solved my issue way before the new versions and could not reproduce the issue anymore.
